src/components/Carousel/CarouselDots.tsx

```diff
@@ -0,0 +1,124 @@
+import classNames from 'classnames';
+import { EmblaCarouselType } from 'embla-carousel';
+import { ComponentProps, forwardRef, useCallback, useEffect, useState } from 'react';
+import { useCarousel } from './utils';
+import { getCommonProps } from '../../utils';
+
+export interface CarouselDotsProps extends ComponentProps<'div'> {
+  /**
+   * The maximum number of dots to display.
+   */
+  maxDots?: number;
+  /**
+   * The position of the dots.
+   */
+  position?: 'on-content' | 'inline';
+}
+
+/**
+ * ## Overview
+ *
+ * Carousel dots component that displays pagination for the carousel.
+ * It supports a maximum number of visible dots
+ * The dots animate left to right as the user navigates through the carousel.
+ *
+ */
+const CarouselDots = forwardRef<HTMLDivElement, CarouselDotsProps>(
+  ({ className, maxDots = 7, position = 'inline', ...props }, ref) => {
```

Comment:
if you have 20 images, and maxDots is set to 7, when you get to the edge of the dots, what happens?  how can you go from image 7 to image 8 with dots?